### ISSUE TITLE
Deprecated feature: REMAKE_INITRD on dkms install on Fedora 35

### ISSUE THREAD

**Issue by snowsnoot** [openzfs/zfs#12867]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Fedora
Distribution Version	| 35
Kernel Version	| 5.15.7-200.fc35.x86_64
Architecture	| x86_64
OpenZFS Version	| zfs-2.1.2-1

### Describe the problem you're observing
When installing from rpm built from source, dkms complains about "Deprecated feature: REMAKE_INITRD". DKMS version is:
```
# rpm -qva | grep dkms
dkms-3.0.2-1.fc35.noarch
zfs-dkms-2.1.2-1.fc35.noarch
```
### Describe how to reproduce the problem
Install ZFS 2.1.2-1 from rpm's built from source. Run 'dkms status zfs' after installing.
### Include any warning/errors/backtraces from the system logs

```
# dkms status zfs
Deprecated feature: REMAKE_INITRD
zfs/2.1.2, 5.15.7-200.fc35.x86_64, x86_64: installedDeprecated feature: REMAKE_INITRD
```

**Comment by xgiovio** [comment]
same problem con centos7 aftter upgrading kernel to 5.15 and system packages. I think it's something new from the dkms package

**Comment by behlendorf** [comment]
This looks like a duplicate of #12781, we'll want to cherry pick the recent fix for the next tagged release.

**Comment by PowerCoder** [comment]
Wanted to add that it also happens on Arch Linux, I'm assuming it's distribution independent.

I now also get:
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead

It could be my dkms package simply needed an update, it was just updated to 3.0.3 on the same upgrade run. If so, feel free to ignore. I figured I would report it since this is the kind of thing which can turn an updating session into an unreadable system.

**Comment by em-winterschon** [comment]
This also occurs on RHEL/derivatives 8.x using kernel-ml versions and OpenZFS 2.1.4. 
```
» dkms autoinstall -m zfs/2.1.4                                                                                                                                                                         
Deprecated feature: REMAKE_INITRD

» dkms status
Deprecated feature: REMAKE_INITRD
zfs/2.1.4, 5.17.3-1.el8.elrepo.x86_64, x86_64: installedDeprecated feature: REMAKE_INITRD

» uname -r
5.17.3-1.el8.elrepo.x86_64

» cat /etc/redhat-release 
Rocky Linux release 8.5 (Green Obsidian)
```
For reference, I'm building ZFS as follows:
```
cd /usr/local/src
wget https://github.com/openzfs/zfs/releases/download/zfs-2.1.4/zfs-2.1.4.tar.gz
extract zfs-2.1.4.tar.gz
cd zfs-2.1.4
./configure --enable-debug
make -s -j$(nproc) rpm-utils rpm-dkms
```

**Comment by behlendorf** [comment]
Thanks for the heads up.  This fix slipped through the cracks for 2.1.4 but it has been added to the 2.1.5 staging branch.

**Comment by SunnyBTC** [comment]
Hi. I'm seeing the same problem on Fedora 5.18.5-100 with zfs 2.1.4. 

Will the staging version 2.1.5 be released anytime soon? If not is there a workaround I can apply to fix the issue?

**Comment by matrais11** [comment]
Same on Debian Testing. I wonder because although Testing is not perfect for sure and bugs might slip in, 'losing' (not seeing) my whole raidz1 pool of 16TB from one day to another is more than annoying. For bleeding edge distros I think it's okay but Testing had at least 1 round of package tests already. If Ubuntu and derivatives (Mint, etc). arrive to this point, they'll have a pain in the a** with this bug. 

Thanks for fixing.

Setting up zfs-dkms (2.1.4-1) ...
Loading new zfs-2.1.4 DKMS files...
Deprecated feature: REMAKE_INITRD (/usr/src/zfs-2.1.4/dkms.conf)
Building for 5.18.0-2-amd64
Building initial module for 5.18.0-2-amd64
Deprecated feature: REMAKE_INITRD (/var/lib/dkms/zfs/2.1.4/source/dkms.conf)
Error! Bad return status for module build on kernel: 5.18.0-2-amd64 (x86_64)
Consult /var/lib/dkms/zfs/2.1.4/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10

**Comment by behlendorf** [comment]
Closing.  This issue was resolved in 2.1.5 which was recently released.

https://github.com/openzfs/zfs/releases/tag/zfs-2.1.5

**Comment by matrais11** [comment]
Sir, You're legend. I see all my pools again (Debian Testing). Love the quick fix, keep up the good work. Respect.
